src/canvas-view/canvas/canvas-viewmodel.js

```diff
@@ -0,0 +1,1154 @@
+    //
+    this.x = function() {
+      return this._x;
+    };
+
+    //
+    // Y coordinate of the connector.
+    //
+    this.y = function() {
+      return this._y;
+    };
+
+    //
+    // The parent node that the connector is attached to.
+    //
+    this.parentNode = function() {
+      return this._parentNode;
+    };
+
+    //
+    // Is this connector connected?
+    //
+    this.connected = function() {
+      return this.data.connected;
+    };
+
+    //
+    // set connector connected
+    //
+    this.setConnected = function(value) {
+      this.data.connected = value;
+    };
+
+    //
+    // Is this connector invalid for a connecton?
+    //
+    this.invalid = function() {
+      return this.data.invalid;
+    };
+
+    //
+    // set connector invalid
+    //
+    this.setInvalid = function(value) {
+      this.data.invalid = value;
+    };
+
+    //
+    // Font Family for the the node.
+    //
+    this.fontFamily = function() {
+      return this.data.fontFamily || "";
+    };
+
+    //
+    // Font Content for the the node.
+    //
+    this.fontContent = function() {
+      return this.data.fontContent || "";
+    };
+  };
+
+  //
+  // Create view model for a list of data models.
+  //
+  var createConnectorsViewModel = function(connectorDataModels, x, parentNode) {
+    var viewModels = [];
+
+    if (connectorDataModels) {
+      for (var i = 0; i < connectorDataModels.length; ++i) {
+        var connectorViewModel = new pfCanvas.ConnectorViewModel(connectorDataModels[i], x, pfCanvas.computeConnectorY(i), parentNode);
+        viewModels.push(connectorViewModel);
+      }
+    }
+
+    return viewModels;
+  };
+
+  //
+  // View model for a node.
+  //
+  pfCanvas.NodeViewModel = function(nodeDataModel) {
+    this.data = nodeDataModel;
+
+    // set the default width value of the node
+    if (!this.data.width || this.data.width < 0) {
+      this.data.width = pfCanvas.defaultNodeWidth;
+    }
+    this.inputConnectors = createConnectorsViewModel(this.data.inputConnectors, 0, this);
+    this.outputConnectors = createConnectorsViewModel(this.data.outputConnectors, this.data.width, this);
+
+    // Set to true when the node is selected.
+    this._selected = false;
+
+    //
+    // Name of the node.
+    //
+    this.name = function() {
+      return this.data.name || "";
+    };
+
+    //
+    // id of the node.
+    //
+    this.id = function() {
+      return this.data.id || -1;
+    };
+
+    //
+    // Image for the the node.
+    //
+    this.image = function() {
+      return this.data.image || "";
+    };
+
+    //
+    // Icon for the the node.
+    //
+    this.icon = function() {
+      return this.data.icon || "";
+    };
+
+    //
+    // Is node a bundle
+    //
+    this.bundle = function() {
+      return this.data.bundle || "";
+    };
+
+    //
+    // background color for the node.
+    //
+    this.backgroundColor = function() {
+      return this.data.backgroundColor || "#bbbbbb";
+    };
```

Comment:
canvas-viewmodel is the original open source canvas code and is defined as a global var which pfCanvas uses.  Unfortunately, canvas-viewmodel isn't compiled as part of the angular flow, so not sure how to inject pfUtils into it.   I could investigate $inject, but this use of #bbbbbb isn't really necessary since we default it to #ffff (see issue above :smile: ), so just removing it.